### PR TITLE
Add polynomial correction to ReflectometryReductionOne

### LINKED ISSUE #11428: Polynomial for Transmission Correction
This issue was originally [TRAC 10586](http://trac.mantidproject.org/mantid/ticket/10586)

Don't do until release 3.4. Ask Owen Arnold for guidance.

Essentially, If no transmission run is provided in the ReflectometryReductionOne workflow algorithms we should:

1) Get a warning about one not being provided. 

2) Create a polynomial correction workspace (similar to the way PolarizationCorrection works) and divide by that.

### PR DESCRIPTION
**Pull request by eXeC64**
Fixes #11428 

Todo:
- [x] Move instrument parameter logic to the `ReflectometryReductionOneAuto` algorithm. All values used in `ReflectometryReductionOne` should be passed as properties.
- [x] Write doc test
- [x] Update documentation to match changes made
- [x] Update release notes
  - [Updates here](http://www.mantidproject.org/ReleaseNotes_3_5_Framework_Changes#Updated_Algorithms)

### CODE REVIEW

**Comment by eXeC64** [comment]
Example for testing:
```python
run = Load(Filename='INTER00013460.nxs')
run2 = CloneWorkspace(run)
SetInstrumentParameter(run2,"polynomial",Value="0,0.5")
IvsQ, IvsLam, thetaOut = ReflectometryReductionOneAuto(InputWorkspace=run, ThetaIn=0.7)
IvsQ2, IvsLam2, thetaOut2 = ReflectometryReductionOneAuto(InputWorkspace=run2, ThetaIn=0.7)
print (IvsLam.readY(0)[3] / IvsLam2.readY(0)[3]) / ((IvsLam.readX(0)[4] + IvsLam.readX(0)[3]) * 0.5)
# 0.5 = (before / after) / (X at centre of bin)
```

**Comment by eXeC64** [comment]
**Testing**:

* Ensure documentation makes sense
* Review code
* Check the new doctest works
* Check polynomial&exponential correction is being run correctly, the python script above is a good starting point.

File: Code/Mantid/Framework/Algorithms/src/ReflectometryReductionOne.cpp [hunk 1]
```diff
@@ -644,6 +679,39 @@ MatrixWorkspace_sptr ReflectometryReductionOne::transmissonCorrection(
 }
 
 /**
+ * Perform Transmission Corrections Using Polynomials.
+ * @param IvsLam : Run workspace which is to be normalized by the results of the
+ * transmission corrections.
+ * @return Normalized run workspace by the generated polynomial.
+ */
+MatrixWorkspace_sptr
+ReflectometryReductionOne::polynomialCorrection(MatrixWorkspace_sptr IvsLam) {
```
Comment (mantid-roman) [inline Code/Mantid/Framework/Algorithms/src/ReflectometryReductionOne.cpp]: The method is called ```polynomialCorrection``` but it does both polynomial and exponential corrections.

File: Code/Mantid/Framework/Algorithms/src/ReflectometryReductionOne.cpp [hunk 2]
```diff
@@ -644,6 +679,39 @@ MatrixWorkspace_sptr ReflectometryReductionOne::transmissonCorrection(
...
+
+  const std::string corrAlgName = getProperty("CorrectionAlgorithm");
+
+  const std::string polynomial = getProperty("Polynomial");
+  const std::string c0 = getProperty("C0");
+  const std::string c1 = getProperty("C1");
```
Comment (mantid-roman) [inline Code/Mantid/Framework/Algorithms/src/ReflectometryReductionOne.cpp]: Is there a reason to translate the property values from their actual types to strings and then translate them back in ```corrAlg```? Are you sure no accuracy is lost?

File: Code/Mantid/Framework/Algorithms/src/ReflectometryReductionOne.cpp [hunk 3]
```diff
@@ -644,6 +679,39 @@ MatrixWorkspace_sptr ReflectometryReductionOne::transmissonCorrection(
...
+
+  IAlgorithm_sptr corrAlg = createChildAlgorithm(corrAlgName);
+  corrAlg->initialize();
+  if (corrAlgName == "PolynomialCorrection") {
+    corrAlg->setProperty("Coefficients", polynomial);
+  } else if (corrAlgName == "ExponentialCorrection") {
+    corrAlg->setProperty("C0", c0);
+    corrAlg->setProperty("C1", c1);
+  } else {
```
Comment (mantid-roman) [inline Code/Mantid/Framework/Algorithms/src/ReflectometryReductionOne.cpp]: I think variables ```polynomial```, ```c0``` and ```c1``` should be declared in the clauses where they are used.

**Pull request merged**